agent/util-scripts/pbench-tool-meister-start

```diff
@@ -531,216 +1082,74 @@ def main(_prog, cli_params):
         optional_md=optional_md,
     )
     try:
-        redis_server.set(tds_param_key, json.dumps(tds, sort_keys=True))
+        redis_client.set(tds_param_key, json.dumps(tds, sort_keys=True))
     except Exception:
         logger.exception(
             "failed to create tool data sink parameter key in redis server"
         )
-        return kill_redis_server(redis_pid, ReturnCode.REDISTDSKEYFAILED)
+        return redis_server.kill(ReturnCode.REDISTDSKEYFAILED)
 
     # +
-    # 4. Start the Tool Data Sink process
+    # Step 4. - Start the Tool Data Sink process (optional)
     # -
 
-    if cli_params.redis_server is None:
-        # FIXME: if only one host is registered, and that host is the same as this
-        # controller, then don't bother starting the Tool Data Sink.
+    if redis_server.managed():
```

Comment:
Should this be checking `tool_data_sink.managed()` instead?  If not, should line 1035 be checking `redis_server.managed()` instead?

(I'm nervous about having two independent checks which are supposed to match each other....)

Reply:
Removed.